### DISCUSSION TITLE
How to register the select tag.

### DISCUSSION THREAD

**Discussion by primado** [react-hook-form/react-hook-form#9963] [Q&A] [answered]
Hello, I have been trying to register the select tag with React Hook From for some days now but it isn't working.
I have been trying to register the select tag for some days now but it isn't working.
I have created a [Code Sandbox](https://codesandbox.io/s/naughty-wright-sgx63z?file=/src/App.js) for this problem. 

Thank you

**Comment by Sajarin-M** [comment, marked as answer]
It seems to be working properly. However, if you're saying that the validation isn't working, it's because the required validation only checks if the value is an empty string, and you're passing a `0` value for the default selected item.
You can either change the first item to something like
```js
<option value=""></option>
```
or you can pass a custom validation rule like
```js
{...register("gender", {
  validate: (value) => (value === "0" ? "Gender is required" : true)
})}
```
if you really want the value to be "0".
